cpp/src/plasma/client.cc

```diff
@@ -72,6 +72,27 @@ constexpr int64_t kThreadPoolSize = 8;
 constexpr int64_t kBytesInMB = 1 << 20;
 static std::vector<std::thread> threadpool_(kThreadPoolSize);
 
+/// A Buffer class that automatically releases the backing plasma object
+/// when it goes out of scope.
+class PlasmaBuffer : public Buffer {
```

Comment:
Does this mean we can get rid of 

https://github.com/apache/arrow/blob/7e27cf511558a92c598bae330958d9c86a721b22/python/pyarrow/plasma.pyx#L182

now?

Reply:
Once we migrate Create() too, yes.